src/features/aws/cognito/authentication/cognito-authentication.ts

```diff
@@ -27,6 +27,29 @@ const tokenExpirationTime = (payload?: string): number => (payload == null ? 0 :
 const substractTimeToTokenExpiration = (token: string, now: Date): number =>
   tokenExpirationTime(getPayload(token)) - now.getTime();
 
+export const getFromJWTPayload = (token: string | null, key: string): Serializable | null => {
+  if (token == null) return null;
+
+  const urlEncodedPayload: string | undefined = token.split('.').at(1);
+  if (typeof urlEncodedPayload !== 'string') return null;
+
+  const base64Payload: string = urlEncodedPayload.replace(/-/gu, '+').replace(/_/gu, '/');
+
+  const jsonPayload: string = decodeURIComponent(
+    atob(base64Payload)
+      .split('')
+      .map((char: string): string => `%${`00${char.charCodeAt(0).toString(16)}`.slice(-2)}`)
```

Comment:
```suggestion
      .map((char: string): string => `%${`00${char.charCodeAt(0).toString(16)}`.slice(-2)}`)
```
Je ne comprends pas ce que ça fait : extraire une fonction nommée